components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java

```diff
@@ -107,6 +113,50 @@ public Response mePost(SelfUserRegistrationRequestDTO selfUserRegistrationReques
         return buildSuccessfulAPIResponse(notificationResponseBean);
     }
 
+    @Override
+    public Response meResendCodePost(MeResendCodeRequestDTO meResendCodeRequestDTO) {
+
+        ResendCodeRequestDTO resendCodeRequestDTO = convertToResendCodeRequest(meResendCodeRequestDTO);
+        NotificationResponseBean notificationResponseBean = null;
+
+        String recoveryScenario = getRecoveryScenarioFromProperties(resendCodeRequestDTO.getProperties());
+        if (!StringUtils.isBlank(recoveryScenario)) {
+            notificationResponseBean = doResendConfirmationCode(recoveryScenario, notificationResponseBean,
+                    resendCodeRequestDTO);
+        }
+
+        if (notificationResponseBean == null) {
+            ErrorDTO errorDTO = Utils.getErrorDTO(Constants.STATUS_BAD_REQUEST_MESSAGE_DEFAULT,
+                    IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_SCENARIO.getCode(),
+                    IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_SCENARIO.getMessage());
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorDTO).build();
+        }
+        return Response.status(Response.Status.CREATED).entity(notificationResponseBean.getKey()).build();
+    }
+
+    @Override
+    public Response meValidateCodePost(MeCodeValidationRequestDTO meCodeValidationRequestDTO) {
+
+        UserSelfRegistrationManager userSelfRegistrationManager = Utils.getUserSelfRegistrationManager();
+        try {
+            // Get the map of properties in the request.
+            HashMap<String, String> propertyMap = Utils.getPropertiesMap(meCodeValidationRequestDTO.getProperties());
+            // Confirm verification code.
+            userSelfRegistrationManager.confirmVerificationCodeMe(meCodeValidationRequestDTO.getCode(), propertyMap);
+        } catch (IdentityRecoveryClientException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Client error while confirming verification code ", e);
```

Comment:
```suggestion
                LOG.debug("Client error while confirming verification code.", e);
```

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/414/commits/b5f0c45e2ce7cb40a24f7a051acc52455755cce4